src/views/ChartPage.vue

```diff
@@ -0,0 +1,463 @@
+      if (masterCategorySelected) {
+        const categories = StoreHandler.getCategoriesByMasterCategory(this.$store, masterCategorySelected, false)
+        for (const category of categories) {
+          switch (type) {
+            case 'allocated':
+              listData.push(Utils.getEurosAmount(this.categoryDataList[category.id]?.allocated))
+              break
+            case 'spent':
+              listData.push(Utils.getEurosAmount(this.categoryDataList[category.id]?.spent * (-1)))
+              break
+            case 'available':
+              listData.push(Utils.getEurosAmount(this.categoryDataList[category.id]?.available))
+              break
+          }
+        }
+      } else {
+        for (const masterCategory of this.$store.state.masterCategories) {
+          let data = 0
+          const categories = StoreHandler.getCategoriesByMasterCategory(this.$store, masterCategory, false)
+          for (const category of categories) {
+            switch (type) {
+              case 'allocated':
+                data += this.categoryDataList[category.id]?.allocated
+                break
+              case 'spent':
+                data += this.categoryDataList[category.id]?.spent * (-1)
+                break
+              case 'available':
+                data += this.categoryDataList[category.id]?.available
+                break
+            }
+          }
+          listData.push(Utils.getEurosAmount(data))
+        }
+      }
+      return listData
+    },
+    getColor (allocated: boolean, spent: boolean, available: boolean, nb: number): string[] {
+      let listColor: string[] = []
+      if (allocated) {
+        listColor.push('#b2babb')
+      }
+      if (spent) {
+        listColor.push('#dc7633')
+      }
+      if (available) {
+        listColor.push('#45c1b8')
+      }
+      if (nb !== 0) {
+        listColor = ['#3498db', '#e74c3c', '#f1c40f']
+        if (nb > 3) {
+          listColor.splice(2, 0, '#27ae60')
+          listColor.push('#fb19cb')
+          if (nb > 5) {
+            listColor.splice(2, 0, '#8e44ad')
+            if (nb > 6) {
+              listColor.splice(2, 0, '#fba619')
+              if (nb > 7) {
+                listColor.splice(6, 0, '#152c8c')
+                listColor.push('#935116', '#fdfefe', '#17202a')
+              }
+            }
+          }
+        }
+      }
+      return listColor
+    },
+    drawPieChart () {
+      const labelList = this.getNames(this.masterCategoryId)
+      const dataList = this.getDatas(this.typeInformationPie, this.masterCategoryId)
+      const colorList = this.getColor(false, false, false, dataList.length)
+      this.pieChartData.labels = labelList
+      this.pieChartData.datasets[0].data = dataList
+      this.pieChartData.datasets[0].backgroundColor = colorList
+    },
+    drawBarChart () {
+      this.chartData.datasets = []
+      if (this.typeInformationBar.indexOf('available') !== -1) {
```

Comment:
could a recursive function on this.typeInformationBar prevent the use of so many if ?

Reply:
done